src/manager.cpp

```diff
@@ -160,13 +181,108 @@ void Manager::deleteSingleFruVpd(
     logging::logMessage(std::string(i_dbusObjPath));
 }
 
+bool Manager::isValidLocationCode(const std::string i_locationCode)
+{
+    if ((i_locationCode.length() < constants::UNEXP_LOCATION_CODE_MIN_LENGTH) ||
+        (i_locationCode[0] != 'U') ||
+        ((i_locationCode.find("fcs", 1, 3) == std::string::npos) &&
+         (i_locationCode.find("mts", 1, 3) == std::string::npos)) ||
+        ((i_locationCode.length() >
+          constants::UNEXP_LOCATION_CODE_MIN_LENGTH) &&
+         (i_locationCode.find("-") != 4)))
+    {
+        return false;
+    }
+
+    return true;
+}
+
 std::string Manager::getExpandedLocationCode(
-    const sdbusplus::message::object_path& i_dbusObjPath)
+    const std::string& i_unexpandedLocationCode,
+    [[maybe_unused]] uint16_t i_nodeNumber = 0)
 {
-    // Dummy code to supress unused variable warning. To be removed.
-    logging::logMessage(std::string(i_dbusObjPath));
+    if (!isValidLocationCode(i_unexpandedLocationCode))
+    {
+        elog<types::InvalidArgument>(
+            types::Argument::ARGUMENT_NAME("LOCATIONCODE"),
+            types::Argument::ARGUMENT_VALUE(i_unexpandedLocationCode.c_str()));
+    }
 
-    return std::string{};
+    nlohmann::json l_parsedJson = m_worker->getJSONObject();
+    if (!l_parsedJson.contains("frus"))
+    {
+        throw JsonException("Missing frus section in JSON", l_parsedJson);
```

Comment:
Why this exception is being thrown over Dbus? For any internal error, handle it internally.

Reply:
Updated